src/testing/deploy/upgrade_test.go

```diff
@@ -123,45 +156,61 @@ func TestUpgradeTrigger(t *testing.T) {
 			for i := 0; i < 11; i++ {
 				require.NoError(t, c.PutFile(dataCommit, "/hello", strings.NewReader("hello world")))
 			}
-			latestDataCI, err := c.InspectCommit(pfs.DefaultProjectName, dataRepo, "master", "")
 			require.NoError(t, err)
+			expectedCommitCount := getExpectedCommitCountFromVersion(from)
 			require.NoErrorWithinTRetry(t, 2*time.Minute, func() error {
-				ci, err := c.InspectCommit(pfs.DefaultProjectName, "TestTrigger2", "master", "")
+				commits, err := c.ListCommit(client.NewRepo(pfs.DefaultProjectName, "TestTrigger1"), nil, nil, 0)
 				require.NoError(t, err)
-				aliasCI, err := c.InspectCommit(pfs.DefaultProjectName, dataRepo, "master", ci.Commit.Id)
 				if err != nil {
 					return err
 				}
-				if aliasCI.Commit.Id != latestDataCI.Commit.Id {
+				fmt.Printf("comparing commit trigger1 sizes %d/%d", len(commits), expectedCommitCount.preTrigger1)
+				if len(commits) < expectedCommitCount.preTrigger1 {
 					return errors.New("not ready")
 				}
 				return nil
 			})
-		},
-		func(t *testing.T, ctx context.Context, c *client.APIClient, _ string) { /* postUpgrade */
```

Comment:
git doesn't do a good job with this diff because of an indentation change from this line in my new changes
```go
			if semver.Compare("v"+from, "v2.8.0") < 0 {
```